components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt

```diff
@@ -43,7 +45,24 @@ class WebExtensionBrowserMenuBuilder(
         val extensionMenuItems =
             WebExtensionBrowserMenu.getOrUpdateWebExtensionMenuItems(store.state, store.state.selectedTab)
 
-        val webExtMenuItem = if (extensionMenuItems.isNotEmpty()) {
+        val filteredExtensionMenuItems = extensionMenuItems.toMutableList()
+
+        val finalList = items.toMutableList()
+
+        extensionMenuItems.forEach { webExtensionBrowserMenuItem ->
+            val index = finalList.indexOfFirst { browserMenuItem ->
+                (browserMenuItem as? PlaceholderBrowserMenuItem)?.label ==
+                    (webExtensionBrowserMenuItem as? WebExtensionBrowserMenuItem)?.id
+            }
+            // replace placeholders with corresponding web extension, and remove it from extensions menu list
+            if (index != -1) {
+                finalList.removeAt(index)
+                finalList.add(index, webExtensionBrowserMenuItem)
```

Comment:
I'm not sure could, but could it cause issue that we looping at the same time that we are editing the same list?

Reply:
`finalList` is not being "looped over" here. So, this should be fine.

Reply:
👍🏽  `finalList` is only looped over when finding the index